### ISSUE TITLE
Device Settings and User-Agent resets to default after login

### ISSUE THREAD

**Issue by alosoft** [adw0rd/instagrapi#316]
**Describe the bug**
am passing ```device_settings``` as well as User-Agent to my server. After successful login with session id. The ```client.device``` returns default settings 
dumping settings to file show default device (Xiaomi) and default User-Agent in json file

**To Reproduce**
Note: ```data``` is dict received from request
```
device_settings = {
        "android_release": "8.1.0",
        "android_version": 27,
        "app_version": "194.0.0.36.172",
        "cpu": "armeabi-v7a",
        "device": "j7popeltetmo",
        "dpi": "280dpi",
        "manufacturer": "samsung",
        "model": "SM-J727T",
        "resolution": "720x1280",
        "version_code": "301484483"
}
```
```
if 'User-Agent' in request.headers:
   client.set_user_agent(request.headers['User-Agent'])

logged_in = False
user_settings = os.path.join(MEDIA_DIR, 'devices', '{}.json'.format(data['userId']))
client = Client()
try:
    if os.path.exists(user_settings):
        client.load_settings(user_settings)
    else:
        if data['device_settings'] is not None:
            client.set_device(device=data['device_settings'])
    logged_in = client.login_by_sessionid(sessionid=session_id)
except AssertionError as asset_error:
    if username and password is None:
        return HttpResponse(asset_error.args, 401)
    else:
        pass

try:
    # check client device and user agent here
    client.dump_settings(user_settings) 
except Exception as error:
    pass

print(client.device)
```
Temporal fix: I set ```client.set_user_device``` before calling ```client.dump_settings```
```
try:
    if 'User-Agent' in request.headers:
        client.set_user_agent(request.headers['User-Agent'])
    if data['device_settings'] is not None:
        client.set_device(device=data['device_settings'])
    client.dump_settings(user_settings)
except Exception as error:
    pass
```

**Desktop (please complete the following information):**
 - OS: Windows
 - instagrapi version 1.9.3

**Comment by adw0rd** [comment]
Fixed in instagrapi 1.9.8

**Comment by alosoft** [comment]
👍
